packages/terra-image/src/Image.jsx

```diff
@@ -47,36 +51,106 @@ const defaultProps = {
   alt: ' ',
 };
 
-const Image = ({
-  src,
-  variant,
-  isFluid,
-  alt,
-  height,
-  width,
-  onLoad,
-  onError,
-  ...customProps
-}) => {
-  const classes = cx([
-    'image',
-    variant,
-    { fluid: isFluid },
-    customProps.className,
-  ]);
-
-  return (
-    <img
-      src={src}
-      alt={alt}
-      height={height}
-      width={width}
-      onLoad={onLoad}
-      onError={onError}
-      {...customProps}
-      className={classes}
-    />);
-};
+class Image extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { isLoading: true, isError: false };
+    const { src, variant, isFluid, alt, placeholder, height, width, onLoad, onError, ...customProps } = this.props;
+
+    this.classes = cx([
+      'image',
+      variant,
+      customProps.className,
+      { fluid: isFluid },
+    ]);
+
+    this.handleOnLoad = this.handleOnLoad.bind(this);
+    this.handleOnError = this.handleOnError.bind(this);
+  }
+
+  componentWillReceiveProps(newProps) {
+    // If a new image is being loaded, reset the state to loading
+    if (newProps.src !== this.props.src) {
+      this.setState({ isLoading: true, isError: false });
+    }
+  }
+
+  handleOnLoad() {
+    this.setState({ isLoading: false });
+    const onLoad = this.props.onLoad;
+    if (onLoad !== undefined) {
+      onLoad();
+    }
+  }
+
+  handleOnError() {
+    this.setState({ isLoading: false, isError: true });
+    const onError = this.props.onError;
+    if (onError !== undefined) {
+      onError();
+    }
+  }
+
+  createPlaceholderImage(customProps) {
+    const { placeholder, alt, height, width } = this.props;
+    return (
+      <img
+        src={placeholder}
+        alt={alt}
+        height={height}
+        width={width}
```

Comment:
What if it errors out here? It feels like we should provide some kind of fallback for this image as well.

Reply:
What do others think about this? It was never discussed, having a fallback for the placeholder

Reply:
If UX doesn't deem it a priority, I'm fine with logging an issue to enhance this later. I think we need to have a universal fallback or all error scenarios, but since we don't have one right now, I fine if we don't address it at the moment.